scripts/quests/sandoria/A_Timely_Visit.lua

```diff
@@ -0,0 +1,208 @@
+-----------------------------------
+-- A Timely Visit
+-----------------------------------
+-- Log ID: 0, Quest ID: 105
+-- Deraquien !pos -98 -2 32 230
+-- Narvecaint !pos -261 23 127 102
+-- Phillone !pos -208 -2 67 230
+-- ??? !pos -310 0 407 104
+-----------------------------------
+local jugnerID = zones[xi.zone.JUGNER_FOREST]
+-----------------------------------
+local fomor = jugnerID.mob.GIOLLEMITTE_B_FEROUN
+local skeleton = jugnerID.mob.GIOLLEMITTE_B_FEROUN + 1
+local quest = Quest:new(xi.questLog.SANDORIA, xi.quest.id.sandoria.A_TIMELY_VISIT)
+
+quest.reward =
+{
+    item     = xi.item.MEDIEVAL_COLLAR,
+    fame     = 60,
+    fameArea = xi.fameArea.SANDORIA,
+    title    = xi.title.OBSIDIAN_STORM,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:getFameLevel(xi.fameArea.SANDORIA) >= 3
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Deraquien'] = quest:progressEvent(33),
+
+            onEventFinish =
+            {
+                [33] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Deraquien'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 0 then
+                        return quest:event(34)
+                    elseif questProgress == 1 then
+                        return quest:progressEvent(80)
+                    elseif questProgress == 2 then
+                        return quest:event(20) -- Additional Dialogue
+                    elseif questProgress == 3 then
+                        return quest:progressEvent(87)
+                    elseif questProgress == 5 then
+                        return quest:progressEvent(38)
+                    end
```

Comment:
What happens when quest progress is 4? default action?

Reply:
Yes